packages/components/src/border-box-control/test/index.js

```diff
@@ -241,35 +291,63 @@ describe( 'BorderBoxControl', () => {
 		} );
 
 		describe( 'Linked value change handling', () => {
-			it( 'should set undefined when new border is empty', () => {
-				renderBorderBoxControl( { value: { width: '1px' } } );
-				updateLinkedWidthInput( '' );
+			it( 'should set undefined when new border is empty', async () => {
+				const user = userEvent.setup( {
+					advanceTimers: jest.advanceTimersByTime,
+				} );
+
+				render(
+					<BorderBoxControl { ...props } value={ { width: '1px' } } />
+				);
+
+				await user.clear( screen.getByRole( 'spinbutton' ) );
 
 				expect( props.onChange ).toHaveBeenCalledWith( undefined );
 			} );
 
-			it( 'should update with complete flat border', () => {
-				renderBorderBoxControl( { value: defaultBorder } );
-				updateLinkedWidthInput( '3' );
+			it( 'should update with complete flat border', async () => {
+				const user = userEvent.setup( {
+					advanceTimers: jest.advanceTimersByTime,
+				} );
+
+				render(
+					<BorderBoxControl { ...props } value={ defaultBorder } />
+				);
+
+				const widthInput = screen.getByRole( 'spinbutton' );
```

Comment:
It would even better if the `spinbutton` had a `name` that we could use in the query (like we do for other buttons).

This change would apply for all of the `screen.getByRole( 'spinbutton' )` queries throughout the file

Reply:
Thanks, that's a good call! I've addressed that in https://github.com/WordPress/gutenberg/pull/45208/commits/21ad8ad92cfa8f9dd9e3af5bed7e851631b757b0 and actually did it for a few more of the elements we're querying. Looks much better now! 🙌